index.js

```diff
@@ -164,24 +179,10 @@ module.exports.read = function(filebuffer, options, fn) {
         fn = fn || options
         options = {}
     }
-    if(!fn || typeof fn !== 'function') {
-        if(typeof filebuffer === "string" || filebuffer instanceof String) {
-            filebuffer = fs.readFileSync(filebuffer)
-        }
-        return this.getTagsFromBuffer(filebuffer, options)
-    } else {
-        if(typeof filebuffer === "string" || filebuffer instanceof String) {
-            fs.readFile(filebuffer, function(err, data) {
-                if(err) {
-                    fn(err, null)
-                } else {
-                    fn(null, this.getTagsFromBuffer(data, options))
-                }
-            }.bind(this))
-        } else {
-            fn(null, this.getTagsFromBuffer(filebuffer, options))
-        }
+    if(isFunction(fn)) {
+        return readAsync.bind(this)(filebuffer, options, fn)

```

Comment:
I am not super keen by this `bind(this)`, this is a **hack** as `readAsync` is not part of the same namespace that it is bound to since `readAsnync` and `readSync` are not exported.

Addressing this properly, would require to not export directly all the functions (same I did for `removeTagsFromBuffer`). It makes it a bit harder to see what is exported when scattered all other the file.
In that case  a single export at the end with all the exported would probably be nicer.

Reply:
There is no state in the NodeID3 "class", so it would make sense to just define all functions without module.exports and then export the public ones at the end so it's not mixed up.

Reply:
OK, I'll do that later this evening.

Reply:
I pushed a massive commit with the changes, it required quite a lot of changes, but this is way better now.
Tests are passing, I had to fix one test.